src/Collection2.java

```diff
@@ -0,0 +1,38 @@
+import java.util.*;
+import java.util.stream.Stream;
+
+public class Collection2 {
+    static String lyrics = Lyrics.lyricsOfYellowSubmarine;
+
+    public static void main(String[] args) {
+        String preparedLyrics = Collection1.preparedString(lyrics);
+        String[] preparedArray = Collection1.arrayOfLyrics(preparedLyrics);
+        int wordCount = arrayToList(preparedArray).size();
+        System.out.println("This list contains "+wordCount+" words");
+        List<String> list = distinctiveList(arrayToList(preparedArray));
+        sortList(list);
+        for (String word: list ){
+            System.out.println(word);
+        }
+    }
+    static List<String> arrayToList(String[] array){
+        return Arrays.asList(array);
+    }
+    static List<String> distinctiveList(List <String> list){
+        Stream<String> stream = list.stream().distinct();
+       return new ArrayList<>(stream.toList());
+        // In previous attempt I accidentally created immutable list and sorting wouldn't work, that's way I initialize new List.
```

Comment:
This method is correctly using a stream to remove duplicates. However, as you noted, using a Set would be more efficient for this purpose, as it automatically ensures that each element is unique.
But you should also remember that Set doesn't maintain the order of elements. As order is important, you might want to use a LinkedHashSet. For this task you don't have to change to set, you can leave it as it is